tests/test_s3.py

```diff
@@ -921,7 +951,10 @@ def test_policy(self):
         session_factory = self.replay_flight_data('test_s3_policy_statements')
 
         client = session_factory().client('s3')
-        client.create_bucket(Bucket=bname)
+        if self.recording:
+            destroyBucketIfPresent(client, bname)
```

Comment:
it would be nicer to cleanup the tests that are leaving turds behind rather then defensively coding around each usage of s3.

Reply:
This one happens to be @whit537 's code, but I'm pretty sure I've done something similar in other tests.  It's actually doing both.  It cleans up after itself, but it's coded defensively so that if something was left behind from a previous run it doesn't fail.

Reply:
I would almost rather not do something like this, just in case someone accidentally runs this against an account where the test bucket might actually legitimately exist.  Failing to create should wake up the tester that something might be wrong and to look into it, instead of just blindly deleting the bucket, not really knowing it's a test bucket or not.

Reply:
after some reflection, i don't think there's a reasonable middle ground option here, functional tests are necessarily destructive, and running them against a live account by accident is something we can only do so much to protect against and still have them run by automation, ie. i expect there are nesc. delays and failures in real world api usage, which need this sort of cleanup, like say the s3 api is down during a test run.